escheduler-server/src/main/java/cn/escheduler/server/worker/task/http/HttpTask.java

```diff
@@ -0,0 +1,269 @@
+        this.httpParameters = JSONObject.parseObject(taskProps.getTaskParams(), HttpParameters.class);
+
+        if (!httpParameters.checkParameters()) {
+            throw new RuntimeException("http task params is not valid");
+        }
+    }
+
+    @Override
+    public void handle() throws Exception {
+        String threadLoggerInfoName = String.format("TaskLogInfo-%s", taskProps.getTaskAppId());
+        Thread.currentThread().setName(threadLoggerInfoName);
+
+        long startTime = System.currentTimeMillis();
+        String statusCode = null;
+        String body = null;
+        boolean status = true;
+        try {
+            CloseableHttpResponse response = sendRequest();
+            statusCode = String.valueOf(getStatusCode(response));
+            body = getResponseBody(response);
+            response.close();
+            exitStatusCode = validResponse(body, statusCode);
+            long costTime = System.currentTimeMillis() - startTime;
+            status = StringUtils.isEmpty(output);
+            logger.info("startTime: {}, httpUrl: {}, httpMethod: {}, status : {}, costTime : {}Millisecond, statusCode : {}, body : {}, log : {}",
+                    DateUtils.format2Readable(startTime), httpParameters.getUrl(),httpParameters.getHttpMethod(), status, costTime, statusCode, body, output);
+        } catch (Exception e) {
+            appendMessage(e.toString());
+            exitStatusCode = -1;
+            logger.error("httpUrl[" + httpParameters.getUrl() + "] connection failed："+output, e);
+        }
+    }
+
+    protected CloseableHttpResponse sendRequest() throws IOException {
+        RequestBuilder builder = createRequestBuilder();
+        ProcessInstance processInstance = processDao.findProcessInstanceByTaskId(taskProps.getTaskInstId());
+
+        Map<String, Property> paramsMap = ParamUtils.convert(taskProps.getUserDefParamsMap(),
+                taskProps.getDefinedParams(),
+                httpParameters.getLocalParametersMap(),
+                processInstance.getCmdTypeIfComplement(),
+                processInstance.getScheduleTime());
+        List<HttpProperty> httpPropertyList = new ArrayList<>();
+        if(httpParameters.getHttpParams() != null && httpParameters.getHttpParams().size() > 0){
+            for (HttpProperty httpProperty: httpParameters.getHttpParams()) {
+                String jsonObject = JSONObject.toJSONString(httpProperty);
+                String params = ParameterUtils.convertParameterPlaceholders(jsonObject,ParamUtils.convert(paramsMap));
+                logger.info("http request params：{}",params);
+                httpPropertyList.add(JSONObject.parseObject(params,HttpProperty.class));
+            }
+        }
+        addRequestParams(builder,httpPropertyList);
+        HttpUriRequest request = builder.setUri(httpParameters.getUrl()).build();
+        setHeaders(request,httpPropertyList);
+        CloseableHttpClient client = createHttpClient();
```

Comment:
Can this be closed. For example, move this output to handle() and don't forget to close